### DISCUSSION TITLE
输入法框架和界面分层

### DISCUSSION THREAD

**Discussion by xingwozhonghua126** [fcitx/fcitx5#352] [Ideas]
输入法包含很多模块，但是界面部分在某些输入法插件（闭源）中希望可以自定义，开源的又期望可以使用默认的，所以希望可能把界面部分和框架拆解出来。

界面部分又分为输入法候选词框，输入法主窗口（状态栏），任务栏菜单，配置界面。最好把开源的模块分细一点。因为任务栏菜单通常要保持统一，候选词每个输入法都想改。

因为新的插件加载机制，开源的可以直接引用闭源的插件。

闭源的输入法可以用自己的。

**Comment by wengxt** [comment]
首先这个方法是不怎么可行的，X11 还可以，但是 wayland 就几乎不行。建议尽量使用 fcitx 内置界面的框架。
https://www.csslayer.info/wordpress/fcitx-dev/client-side-input-panel-for-fcitx-5/

只说 X11 的话，如果闭源的 UI 不通用（指无法支持其他的输入法），让输入法引擎部分和界面去交互就行了，自然就没有界面。
一句话，不要用 Fcitx 的 UI 部分的框架和代码，完全自己处理即可。
